The-SMIRNOFF-force-field-format.md

```diff
@@ -1,103 +1,288 @@
+* `Coulomb` - direct Coulomb interactions (with no reaction-field attenuation) should be used
+
+The interaction scaling parameters applied to atoms connected by a few bonds are
+* `scale12` (default: 0) specifies the scaling applied to 1-2 bonds
+* `scale13` (default: 0) specifies the scaling applied to 1-3 bonds
+* `scale14` (default: 0.833333) specifies the scaling applied to 1-4 bonds
+
+Currently, no child tags are used because the charge model is specified via different means (currently library charges or BCCs).
+
+For methods where the cutoff is not simply an implementation detail but determines the potential energy of the system (`reaction-field` and `Coulomb`), the `cutoff` distance must also be specified, and a `switch-width` if a switching function is to be used.
+
+**TODO:** Also add the dispersion correction behavior to `StericsForce`.
+
+**QUESTION:** I've opted to not make `<LibraryCharges>` and `<BondChargeCorrections>` be sub-tags of `<ElectrostaticsForce>` since, while this may logically make sense, it makes the code *much* harder to make truly modular. For example, adding a polarizable force would require we modify the original code inside the `openforcefield` toolkit, rather than simply create a new experimental `ForceGenerator` outside the toolkit and import it for experimentation. Is this OK, or will it lead to logical problems?
+
+### `<BondForce>`
+
+Bond parameters are specified via a `<BondForce>...</BondForce>` block, with individual `<Bond/>` tags containing attributes specifying the equilibrium bond length (`length`) and force constant (`k`) values for specific bonds.
+For example:
+```XML
+<BondForce potential="harmonic" length_unit="angstroms" k_unit="kilocalories_per_mole/angstrom**2">
    <Bond smirks="[#6X4:1]-[#6X4:2]" length="1.526" k="620.0"/>
    <Bond smirks="[#6X4:1]-[#1:2]" length="1.090" k="680.0"/>
-...
-</HarmonicBondForce>
+   ...
+</BondForce>
 ```
+Currently, only `potential="harmonic"` is supported, where we utilize the standard harmonic functional form:
+```
+U(r) = (k/2)*(r-length)^2
+```
+Later revisions will add support for additional potential types and the ability to support arbitrary algebraic functional forms.
+If the `potential` attribute is omitted, it defaults to `harmonic`.
 
-**AMBER functional forms define the force constant `k` in a manner that differs by a factor of two---we do not use that convention here, electing to use the standard harmonic definition `U(r) = (k/2)*(r-length)^2` instead.**
-Thus, comparing a SMIRNOFF file to a corresponding AMBER parameter file or `.frcmod` will make it appear that force constants here are twice as large as they ought to be.
+**Note that AMBER and CHARMM define a modified functional form**, such that `U(r) = k*(r-length)^2`, so that force constants would need to be multiplied by two in order to be used in the SMIRNOFF format.
```

Comment:
To be explicit, does this mean, if I go to my `leaprc.ff14SB` and see the line:
```
C -2C  317.0    1.5220
```
...for some carbon-carbon bond, and I decide I want to add this parameter to my SMIRNOFF-based FF, that I should *multiply* the `317.0` by two to put this in SMIRNOFF format?

Reply:
Yes. Isn't that what it says? (If it's not clear maybe suggest a rephrasing, @slochower ?)

Reply:
Ha, yes I guess that is exactly what it says. Just confirming.